### PR TITLE
Qual 569 port to v2 approvals api

### PR DESCRIPTION
**Pull request by CodeWolf86**
Navigate to here: https://dev-tools.apprenticeships.education.gov.uk/home

And select Support Tool to see this PR's version of the support tool:
- Removes original form to enter ID's directly.
- Adds in search mechanism against das-commitments api with an endpoint removing the search constraints
- Uses those search results to pick an apprenticeship to stop
- Stops the apprenticeship targeting new V2 Endpoint

Das-Commitments PR available here: https://github.com/SkillsFundingAgency/das-commitments/pull/769

### CODE REVIEW

File: src/SFA.DAS.Tools.Support.Web/Views/Approvals/SearchApprenticeships.cshtml [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+                        Submit
+                    </button>
+                </div>
+            </form>
+        </div>
+    </div>
+</div>
+<div id="searchResults" style="display:none">
+    <table id="apprenticeshipResultsTable"
+           data-toggle="table"
```
Comment (DannyGB) [inline src/SFA.DAS.Tools.Support.Web/Views/Approvals/SearchApprenticeships.cshtml:112]: add in data-search so the data in the table can be searched

File: src/SFA.DAS.Tools.Support.Web/Views/Approvals/StopApprenticeship.cshtml [hunk 2]
```diff
@@ -4,61 +4,90 @@
...
-                <input asp-for="ApprenticeshipId" id="apprenticeshipId" class="govuk-input govuk-input--width-20" />
-            </div>
-            <div class="govuk-form-group">
-                <label asp-for="EmployerAccountId" class="govuk-label">
-                    EmployerAccount Id
-                </label>
-                <span id="EmployerAccountId" class="govuk-error-message">
-                    @Html.ValidationMessageFor(m => m.EmployerAccountId)
-                </span>
+@if (Model != null && Model.ApprenticeshipNotFound)
```
Comment (DannyGB) [inline src/SFA.DAS.Tools.Support.Web/Views/Approvals/StopApprenticeship.cshtml:7]: As per our earlier conversation we should change this page when we do the bulk change task to allow bulk updates

**Review by DannyGB — APPROVED** [review]

File: src/SFA.DAS.Tools.Support.Infrastructure/Services/EmployerCommitmentsService.cs [hunk 3]
```diff
@@ -61,28 +74,83 @@ public async Task<StopApprenticeshipResult> StopApprenticeship(long employerAcco
...
                 if (apprenticeshipId <= 0)
                 {
-                    throw new ArgumentException("apprenticeshipId must be greater than 0", "apprenticeshipId");
+                    throw new ArgumentException("ApprenticeshipId must be greater than 0", "apprenticeshipId");
                 }
 
-                var result = await _employerCommitmentApi.GetEmployerApprenticeship(employerAccountId, apprenticeshipId);
+                var result = await _commitmentApi.GetApprenticeship(apprenticeshipId, token);
+
+                var r =new GetApprenticeshipResult
```
[the remark this answers is not in the document]
Comment (CodeWolf86) [reply src/SFA.DAS.Tools.Support.Infrastructure/Services/EmployerCommitmentsService.cs:134]: Nice catch, think I was doing some debugging and left it in. Have updated.

File: src/SFA.DAS.Tools.Support.Web/App_Start/IoC.cs [hunk 4]
```diff
@@ -1,36 +1,31 @@
...
+                config.AddProfile<AutoMapperProfile>();
             }, typeof(Startup));
-            services.AddScoped<IEmployerCommitmentApi, EmployerCommitmentApi>(provider =>
+            services.AddSingleton<ICommitmentsApiClientFactory>(x =>
             {
-                var config = _configuration.GetSection("CommitmentsApiClientConfiguration").Get<CommitmentsApiClientConfiguration>();
-                var httpClient = new HttpClientBuilder().WithBearerAuthorisationHeader(new AzureActiveDirectoryBearerTokenGenerator(config)).Build();
-                return new EmployerCommitmentApi(httpClient, config);
+                var config = x.GetService<IOptions<CommitmentsClientApiConfiguration>>().Value;
+                var loggerFactory = x.GetService<ILoggerFactory>();
```
[the remark this answers is not in the document]
Comment (CodeWolf86) [reply src/SFA.DAS.Tools.Support.Web/App_Start/IoC.cs:25]: Done

File: src/SFA.DAS.Tools.Support.Web/Controllers/ApprovalsController.cs [hunk 5]
```diff
@@ -24,69 +30,111 @@ public ApprovalsController(ILogger<ApprovalsController> logger, IEmployerCommitm
...
+                {
+                    Id = a.Id,
+                    FirstName = a.FirstName,
+                    LastName = a.LastName,
+                    EmployerName = a.EmployerName,
+                    ProviderName = a.ProviderName,
+                    StartDate = a.StartDate.ToShortDateString(),
+                    EndDate = a.EndDate.ToShortDateString(),
+                    Status = a.ApprenticeshipStatus.ToString()
+                }));
```
[the remark this answers is not in the document]
Comment (CodeWolf86) [reply src/SFA.DAS.Tools.Support.Web/Controllers/ApprovalsController.cs:76]: Fair point. I reworked this code a number of times when sorting bootstrap so missed the simplification. Updated.

File: src/SFA.DAS.Tools.Support.Web/Views/Approvals/SearchApprenticeships.cshtml (hunk 1 above)
Comment (CodeWolf86) [reply src/SFA.DAS.Tools.Support.Web/Views/Approvals/SearchApprenticeships.cshtml:112]: Added

File: src/SFA.DAS.Tools.Support.Web/Views/Approvals/StopApprenticeship.cshtml (hunk 2 above)
Comment (CodeWolf86) [reply src/SFA.DAS.Tools.Support.Web/Views/Approvals/StopApprenticeship.cshtml:7]: Will add this to the bulk updates task 👍

**Pull request merged**